query/sources/common/src/main/java/org/codice/ddf/admin/sources/services/CswServiceProperties.java

```diff
@@ -0,0 +1,144 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.collect.BiMap;
+import com.google.common.collect.ImmutableBiMap;
+import com.google.common.collect.ImmutableList;
+
+public class CswServiceProperties {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(CswServiceProperties.class);
+
+    public static final String CSW_URL = "cswUrl";
+
+    public static final String EVENT_SERVICE_ADDRESS = "eventServiceAddress";
+
+    public static final String OUTPUT_SCHEMA = "outputSchema";
+
+    public static final String FORCE_SPATIAL_FILTER = "forceSpatialFilter";
+
+    public static final String CSW_PROFILE_FACTORY_PID = "Csw_Federation_Profile_Source";
+
+    public static final String CSW_GMD_FACTORY_PID = "Gmd_Csw_Federated_Source";
+
+    public static final String CSW_SPEC_FACTORY_PID = "Csw_Federated_Source";
+
+    public static final String USERNAME = "username";
+
+    public static final String PASSWORD = "password";
+
+    public static final String ID = "id";
+
+    public static final List<String> CSW_FACTORY_PIDS = ImmutableList.of(CSW_PROFILE_FACTORY_PID,
+            CSW_GMD_FACTORY_PID,
+            CSW_SPEC_FACTORY_PID);
+
+    public static final Function<Map<String, Object>, SourceConfigUnionField>
+            SERVICE_PROPS_TO_CSW_CONFIG = CswServiceProperties::servicePropsToCswConfig;
+
+    private static final BiMap<String, String> CSW_PROFILE_MAPPING = ImmutableBiMap.of(
+            CSW_FEDERATION_PROFILE_SOURCE, CSW_PROFILE_FACTORY_PID,
+            GMD_CSW_ISO_FEDERATED_SOURCE, CSW_GMD_FACTORY_PID,
+            CSW_SPEC_PROFILE_FEDERATED_SOURCE, CSW_SPEC_FACTORY_PID);
+
+    public static final CswSourceConfigurationField servicePropsToCswConfig(
+            Map<String, Object> props) {
+        CswSourceConfigurationField cswConfig = new CswSourceConfigurationField();
+        cswConfig.pid(mapStringValue(props, SERVICE_PID_KEY));
+        cswConfig.sourceName(mapStringValue(props, ID));
+        cswConfig.endpointUrl(mapStringValue(props, CSW_URL));
+        cswConfig.credentials()
+                .username(mapStringValue(props, USERNAME));
+        cswConfig.credentials()
+                .password(mapStringValue(props, PASSWORD));
+        cswConfig.outputSchema(mapStringValue(props, OUTPUT_SCHEMA));
+        cswConfig.spatialOperator(mapStringValue(props, FORCE_SPATIAL_FILTER));
+        try {
+            cswConfig.cswProfile(factoryPidToCswProfile(mapStringValue(props, FACTORY_PID_KEY)));
+        } catch (IllegalArgumentException ignored) {
+        }
+        return cswConfig;
+    }
+
+    public static final Map<String, Object> cswConfigToServiceProps(
+            CswSourceConfigurationField config) {
+        HashMap<String, Object> props = new HashMap<>();
+        props.put(ID, config.sourceName());
+        props.put(CSW_URL, config.endpointUrl());
+        if (config.eventServiceAddress() != null) {
```

Comment:
If we provide a null value inside the map, that service property's value will be overwritten. For instance, if we create a source config with a username/password and then update it without sending the username/password, then those service names will be overwritten. We should actually be null checking all of these.